impacted-test-engine/src/main/java/com/teamscale/test_impacted/engine/TestDataWriter.java

```diff
@@ -33,10 +34,10 @@ void dumpTestExecutions(List<TestExecution> testExecutions) {
 	}
 
 	/** Writes the given test details to a report file. */
-	void dumpTestDetails(List<TestDetails> testDetails) {
+	void dumpTestDetails(List<? extends TestDetails> testDetails) {
 		File file = new File(reportDirectory, "test-list.json");
 		try {
-			ReportUtils.writeTestListReport(file, testDetails);
+			ReportUtils.writeTestListReport(file, new ArrayList<>(testDetails));
```

Comment:
Because the given list has type `List<? extends TestDetails>` and the method only accepts `List<TestDetails>` and besides a force cast this was the only way to safely convert the list's type